src/admin/containers/email.js

```diff
@@ -22,34 +23,37 @@ const Email = (
   , send_newsletter: news
   , send_newsletter_reminder: remind
   , compose_custom: custom
-  , emails
-  , custom_emails
-  , submit_email
+  , get_bounced
+  , active_tab
   , email_sent
-  , submit_custom_email
   , ...list_props
   }
 ) =>
   <div className='main-container email'>
-    <form
-      className='email-controls'
-    >
-      { map(send_button, zip(email_ids, [sub, news, remind, custom])) }
+    <form className='email-controls' >
+    { map(send_button, zip(email_ids, [sub, news, remind, custom, get_bounced])) }
     </form>
+
     {email_sent
-      ? <EmailNotification email_sent={email_sent}/>
-      : (custom_emails
-            ? <CustomEmailForm submit={submit_custom_email} members={custom_emails.members} email_sent={email_sent} />
-            : keys(emails).length > 0 && email_list({ emails, submit_email, ...list_props })
-        )
+      ? <EmailNotification email_sent={email_sent} />
+      : active_tab && map_tab({...list_props})[active_tab]()
     }
-  </div>
+    </div>
+
+const map_tab = props => (
```

Comment:
I would rewrite this to be

```js
{ SEND_SUB_REMINDER: email_list
, SEND_NEWSLETTER: email_list
, SEND_NEWSLETTER_REMINDER: email_list
, COMPOSE_CUSTOM: props => <CustomEmailForm submit={props.submit_custom_email} members={props.custom_emails.members}/>
, GET_BOUNCED: BouncedEmails
}
```

One could get rid of some of the duplication with `cond`, but I think it'd be OK like this